codebuild/bin/s2n_open_fds_test.py

```diff
@@ -0,0 +1,65 @@
+# Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+# This script parses the LastDynamicAnalysis file generated by CTest Memcheck.
+# It identifies any leaking file descriptors and triggers an error when detected.
+# This enhances the capabilities of existing Valgrind checks.
+# Output snippet for open file descriptors:
+# FILE DESCRIPTORS: 6 open (3 std) at exit.
+#  ==6652== Open AF_INET socket 6: 127.0.0.1:36915 <-> unbound
+#  ==6652==    at 0x498B2EB: socket (syscall-template.S:120)
+#  ==6652==    by 0x16CD16: s2n_new_inet_socket_pair (s2n_self_talk_ktls_test.c:69)
+#  ==6652==    by 0x15DBB2: main (s2n_self_talk_ktls_test.c:168)
+#  ==6652==
+import os
+import sys
+
+# Exit with error code 1 if leaking fds are detected.
+ERROR_EXIT_CODE = 1
+# This test is designed to be informational, so we only print five lines when a leak is detected.
+NUM_OF_LINES_TO_PRINT = 5
+
+
+def open_analysis_file(path):
+    for fl in os.listdir(path):
+        if "LastDynamicAnalysis" in fl:
+            file = open(os.path.join(path, fl), 'r')
+    return file
+
+
+def read_analysis_file(file):
+    exit_code = 0
+    lines = file.readlines()
+    for i in range(len(lines)):
+        if "FILE DESCRIPTORS:" in lines[i]:
+            line_elements = lines[i].split()
+            # The element after "FILE DESCRIPTORS:" is the number of fds that remain opened at exit.
+            open_fd_number = line_elements[line_elements.index("DESCRIPTORS:") + 1]
```

Comment:
> I'd prefer to directly index to the token.

Direct indexing can't handle edge cases. There are many tests which fork many processes during execution, and the first `FILE DESCRIPTORS:` will show up like
```
Running /codebuild/output/src1744391194/src/tests/unit/s2n_fragmentation_coalescing_test.c ... ==6204== FILE DESCRIPTORS: 4 open (3 std) at exit.
```
Hence, we can't predict where exactly the token is located. That is also why I used `.index` to locate those tokens that we need.

> I think the commenting also might be a bit overboard, as the variables names pretty clearly explain what each of these items are.

I think the first comment is probably pretty clear, so I can remove that. However, I would still keep the next two comments, since they explain why we are taking those actions. 

> Although I'd vote to rename open_fd_number to open_fd_count since it's a bit more specific.

I will do this.